src/jdk.jshell/share/classes/jdk/internal/jshell/tool/ConsoleIOContext.java

```diff
@@ -977,7 +977,15 @@ public void perform(LineReaderImpl in) throws IOException {
     public synchronized int readUserInput() throws IOException {
         if (pendingBytes == null || pendingBytes.length <= pendingBytesPointer) {
             char userChar = readUserInputChar();
-            pendingBytes = String.valueOf(userChar).getBytes();
+            StringBuilder dataToConvert = new StringBuilder();
```

Comment:
Perhaps, add here the comment from the PR description for readers from the future:

> [...] when the current character is a high surrogate, peek at the next character, and if it is a low surrogate, convert both the high and low surrogates to bytes together.

The (internal) API used in the implementation doesn't express that on first sight.

Reply:
Thanks for adding comments.